_javascripts/page-loader.js

```diff
@@ -1,114 +1,68 @@
-// the new final link to load
-newLink = "";
-newVersion = "";
-currentVersion = "";
-
-// the fileRequested
-fileRequested = "";
-
-//Array prototype to check if an element is in an array
-Array.prototype.contains = function(obj) {
-  return this.indexOf(obj) > -1;
+let newLink = "";
+let newVersion = "";
+let currentVersion = "";
+let fileRequested = "";
+
+const urlMappings = {
+  "openshift-acs": "https://docs.openshift.com/acs/",
+  "openshift-builds": "https://docs.openshift.com/builds/",
+  "openshift-enterprise": "https://docs.openshift.com/container-platform/",
+  "openshift-gitops": "https://docs.openshift.com/gitops/",
+  "openshift-origin": "https://docs.okd.io/",
+  "openshift-pipelines": "https://docs.openshift.com/pipelines/",
+  "openshift-serverless": "https://docs.openshift.com/serverless/",
+};
+
+// Helper function to confirm asynchronously
+async function confirmAsync(message) {
+  return window.confirm(message);
 }
 
 function versionSelector(list) {
+  "use strict";
 
-  // the version we want
   newVersion = list[list.selectedIndex].value;
+  currentVersion = window.location.pathname.split("/")[2];
 
-  // spilt the current path
-  var pathArray = window.location.pathname.split( '/' );
-
-  // so we can get the current version
-  currentVersion = pathArray[2];
+  let baseUrl = urlMappings[dk];
 
-  // if switching major versions, just take the user to the main landing page
-  // as files change a lot between major versions.
+  //Handle special OCP case
+  if (["3.0", "3.1", "3.2"].includes(newVersion) && dk === "openshift-enterprise") {
+    baseUrl = "https://docs.openshift.com/enterprise/";
+  }
 
-  if(currentVersion.charAt(0) === newVersion.charAt(0)) {
-    // the file path is just the version number + the end of the path
-    fileRequested =
-      window.location.pathname.substring(
-        window.location.pathname.lastIndexOf(currentVersion) +
-        currentVersion.length);
+  if (currentVersion.charAt(0) === newVersion.charAt(0)) {
+    const versionIndex = window.location.pathname.lastIndexOf(currentVersion) + currentVersion.length;
+    fileRequested = window.location.pathname.substring(versionIndex);
   } else {
     fileRequested = "/welcome/index.html";
   }
```

Comment:
I did not notice this part. I'm afraid /welcome/index.html won't work for layered products, though I guess this can be fixed with a mandarory redirect (I really need to know how to set up redirects). Moreover, sometimes a page can cease to exist *between* minor versions, so the blanket assumption that the same path works is not always true. People don't have to wait for (very rare) major versions to restructure docs.